### ISSUE TITLE
The average Episodic Return and Average Loss is nan

### ISSUE THREAD

**Issue by britig** [ericyangyu/PPO-for-Beginners#1]
Hi Eric,

I am a beginner with PPO and I tried your code with the _log_summary() function implemented with the following main block.
```
if __name__ == '__main__':
	hyperparameters = {
				'timesteps_per_batch': 2048, 
				'max_timesteps_per_episode': 200, 
				'gamma': 0.99, 
				'n_updates_per_iteration': 10,
				'lr': 3e-4, 
				'clip': 0.2
			  }
	env = gym.make('Pendulum-v0')
	model = PPO(env=env, **hyperparameters)
	print(f'Model information ====== {model}')
	model.learn(10000)
```

The output keeps on giving nan value on Episodic Return and Average Loss. For example

-------------------- Iteration #50 --------------------
Average Episodic Length: 1.0
Average Episodic Return: nan
Average Loss: nan
Timesteps So Far: 50
------------------------------------------------------

Can you kindly help with this? The policy network is FeedForwardNN.

**Comment by ericyangyu** [comment]
Hi Briti!

I just tried your block of code with the repo code, and it works out for me (had to change ```model = PPO(env=env, **hyperparameters)``` to ```model = PPO(policy_class=FeedForwardNN, env=env, **hyperparameters)``` though).

Output: 
```
Model information ====== <ppo.PPO object at 0x7fa61287c358>
Learning... Running 200 timesteps per episode, 2048 timesteps per batch for a total of 1000000 timesteps

-------------------- Iteration #1 --------------------
Average Episodic Length: 200.0
Average Episodic Return: -1358.09
Average Loss: -0.00129
Timesteps So Far: 2200
------------------------------------------------------


-------------------- Iteration #2 --------------------
Average Episodic Length: 200.0
Average Episodic Return: -1172.53
Average Loss: -0.001
Timesteps So Far: 4400
------------------------------------------------------

^C
```

I suspect something in your ```ppo.py``` implementation is incorrect; would you mind sending me your ```ppo.py``` code?

**Comment by britig** [comment]
Hi Eric,

Thank you for your reply. Attaching my
[ppo.zip](https://github.com/ericyangyu/PPO-for-Beginners/files/5633648/ppo.zip)
 ppo.py. I  suspect there is something wrong with the get_action() method. Kindly help.

**Comment by ericyangyu** [comment]
I resolved the nans :) So the issue was actually in `rollout` and `compute_rtgs`; you had some pieces of code on the wrong indent, which were early terminating the two functions before they had a chance to finish their loops.

Specifically in `rollout`, I shifted:
```
# Collect episodic length and Rewards
batch_lens.append(ep_t + 1)
batch_rews.append(ep_rews)
```
one indent left, and 

```
#Reshape data as tensors
batch_obs = torch.tensor(batch_obs, dtype=torch.float)
batch_acts  = torch.tensor(batch_acts, dtype=torch.float)
batch_log_probs = torch.tensor(batch_log_probs, dtype=torch.float)

# ALG STEP 4

batch_rtgs = self.compute_rtgs(batch_rews)

print(f'Batch rewards before logger======={batch_rews}')

# Log the episodic returns and episodic lengths in this batch.
self.logger['batch_rews'] = batch_rews
self.logger['batch_lens'] = batch_lens

# Return the batch data
return batch_obs, batch_acts, batch_log_probs, batch_rtgs, batch_lens
```
two indents left.

In `compute_rtgs`, I shifted:
```
#Convert the reward-to-go into a tensor
batch_rtgs = torch.tensor(batch_rtgs, dtype=torch.float)

return batch_rtgs
```
one indent left.

Here's ppo.py with the fixes: [ppo.py.zip](https://github.com/ericyangyu/PPO-for-Beginners/files/5633874/ppo.py.zip). Let me know if there's anything else I can help with :)

**Comment by britig** [comment]
Works :) Thanks a lot..
